### PR TITLE
Closes #1741 Hide default intelligenceBank Media Type.

### LINKED ISSUE #1741: Remove remotely hosted media from az_digital_asset_library
## Motivation
Users find it to be too complicated and unfriendly to see a new section of media without any indication of how to add media.

**Note:** This is the same behavior as the other tabs (audio, document, image, remote video) that do not have any managed media

### Is your feature request related to a problem? Please describe.
DAM pop up under Media

## Proposed Resolution
Remove functionality for using remotely hosted media.

### PR DESCRIPTION
**Pull request by trackleft**
## Description
Added preprocess functions to hide links for intelligence bank media types in both the media embed dialog and on the add/media page

## Related issues
Closes #1741 
Follow-up PR https://github.com/az-digital/az_quickstart/pull/1752 (Would need to remove changes from this PR)
Waiting on Drupal issue: https://www.drupal.org/project/intelligencebank/issues/3295615

## How to test
Go to any page content
Add text paragraph
Attempt to embed media
See no option for intelligence bank dam media type

Go to /media/add page
See no option for intelligence bank dam media type

Go to /media/add/ib_dam_embed
See available options (DO NOT USE THOUGH)

### Arizona Quickstart (install profile, custom modules, custom theme)
- Patch release changes
   - [x] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
- Minor release changes
   - [ ] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- [ ] Other or unknown

### Drupal core
- Patch release changes
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major or minor level update
- [ ] Other or unknown

### Drupal contrib projects
- Patch release changes
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- Minor release changes
   - [ ] Major level update
- [ ] Other or unknown

## Checklist
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_digital_asset_library/az_digital_asset_library.module [hunk 1]
```diff
@@ -27,3 +27,33 @@ function az_digital_asset_library_form_submit(array $form, FormStateInterface $f
 function az_digital_asset_library_form_entity_browser_az_digital_asset_library_form_alter(&$form, FormStateInterface $form_state, $form_id) {
   $form['#submit'][] = 'az_digital_asset_library_form_submit';
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ *
+ * Hide default provided media type by IntelligenceBank from media select dialog.
```
Comment (joeparsons) [inline modules/custom/az_digital_asset_library/az_digital_asset_library.module:34]: Nitpick:
```suggestion
 * Hide default media type provided by IntelligenceBank from media select dialog.
```

File: modules/custom/az_digital_asset_library/az_digital_asset_library.module [hunk 2]
```diff
@@ -27,3 +27,33 @@ function az_digital_asset_library_form_submit(array $form, FormStateInterface $f
...
+    isset($variables['links']['media-library-menu-ib_dam_embed']['link'])
+    ) {
+    $variables['links']['media-library-menu-ib_dam_embed']['link']['#access'] = FALSE;
+  }
+}
+
+/**
+ * Implements hook_preprocess_HOOK().
+ *
+ * Hide default provided media type by IntelligenceBank from media add page.
```
Comment (joeparsons) [inline modules/custom/az_digital_asset_library/az_digital_asset_library.module:51]: Nitpick:
```suggestion
 * Hide default media type provided by IntelligenceBank from media add page.
```

File: modules/custom/az_digital_asset_library/az_digital_asset_library.module [hunk 3]
```diff
@@ -27,3 +27,33 @@ function az_digital_asset_library_form_submit(array $form, FormStateInterface $f
...
+ *
+ * Hide default provided media type by IntelligenceBank from media select dialog.
+ *
+ * See @link https://github.com/az-digital/az_quickstart/issues/1741
+ */
+function az_digital_asset_library_preprocess_links__media_library_menu(&$variables) {
+  if (
+    isset($variables['links']) &&
+    isset($variables['links']['media-library-menu-ib_dam_embed']) &&
+    isset($variables['links']['media-library-menu-ib_dam_embed']['link'])
```
Comment (bberndt-uaz) [inline modules/custom/az_digital_asset_library/az_digital_asset_library.module:42]: It looks like we don't need a test for each key: 

- See the last line before Example 2: https://www.php.net/manual/en/function.isset.php
- Or [this StackOverflow answer](https://stackoverflow.com/questions/832408/how-to-check-if-a-deep-array-value-is-present)

**Review by joeparsons — APPROVED** [review]
Looks good to me.

**Review by bberndt-uaz — APPROVED** [review]

**Pull request merged**